bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/layers/GridFreezeLayer.java

```diff
@@ -80,6 +86,41 @@ public void createEmptyAnnotationSpan(Set<PositionCoordinate> selectedCoordinate
     bodyDataProvider.createEmptyAnnotationSpan(selectedCoordinates);
   }
 
+  /**
+   * Creates a new column of the given type, with the given qualified annotation name, and adds it
+   * to the list of columns at the given insertion index.
+   * 
+   * @param type The type of the column to be created
+   * @param annoQName The qualified annotation name of the column
+   * @param insertionIndex The index at which the column to be created should be inserted into the
+   *        list of columns, or -1 when it should be added at the end of the list
+   */
+  public void addAnnotationColumn(ColumnType type, String annoQName, int insertionIndex) {
+    int existingColumnCounter = 1;
```

Comment:
```suggestion
    // Counter starts at one as the added column represents the first existing column 
    // for the specified qualified annotation name.
    int existingColumnCounter = 1;
```

Agree, explaining this in a comment.